vllm/distributed/device_communicators/custom_all_reduce_utils.py

```diff
@@ -198,7 +201,27 @@ def gpu_p2p_access_check(src: int, tgt: int) -> bool:
         ids = list(range(num_dev))
         # batch of all pairs of GPUs
         batch_src, batch_tgt = zip(*list(product(ids, ids)))
-        result = can_actually_p2p(batch_src, batch_tgt)
+        # NOTE: we use `subprocess` rather than `multiprocessing` here
+        # because the caller might not have `if __name__ == "__main__":`,
+        # in that case we cannot use spawn method in multiprocessing.
+        # However, `can_actually_p2p` requires spawn method.
+        # The fix is, we use `subprocess` to call the function,
+        # where we have `if __name__ == "__main__":` in this file.
+        input_bytes = pickle.dumps((batch_src, batch_tgt))
+        returned = subprocess.run([sys.executable, __file__],
+                                  input=input_bytes,
+                                  capture_output=True)
+        # check if the subprocess is successful
+        try:
+            returned.check_returncode()
+        except Exception as e:
+            # wrap raised exception to provide more information
+            logger.error(
+                "Error happened when batch testing "
+                "peer-to-peer access"
+                " from %s to %s", batch_src, batch_tgt)
+            raise e
```

Comment:
Was thinking to wrap instead
```suggestion
            raise RuntimeError(f"Error happened when "
                "batch testing peer-to-peer access "
                "from {batch_src} to {batch_tgt}") from e
```

Reply:
fixed in [8348ef5](https://github.com/vllm-project/vllm/pull/5669/commits/8348ef58bb55c57dafcc1e27c53bf98c44254ddd)